tests/unit-tests/platforms/mesa/kms/test_cursor.cpp

```diff
@@ -419,7 +420,9 @@ MATCHER_P(ContainsASingleWhitePixel, buffersize, "")
     auto pixels = static_cast<uint32_t const*>(arg);
     if (pixels[0] != 0xffffffff)
         return false;
-    for (decltype(buffersize) i = 1; i < buffersize; i++)
+    // get a non-const instance of the buffersize type (needed for gtest 1.8.1+)
+    typename std::remove_const<decltype(buffersize)>::type i;
+    for (i = 1; i < buffersize; i++)
```

Comment:
No need to increase the scope of `i`. Vis:

    for (auto i = decltype(buffersize){1}; i < buffersize; i++)